File: samples/python/04primality/prime.py
import math
import sys


def is_prime(number):
    limit = math.sqrt(number)
    start = 2
    while start <= limit:
        if number % start == 0:
            return False
        start += 1
    return True


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Error: Usage python prime.py <number> [<number> ...]")
        sys.exit(1)

    for arg in sys.argv[1:]:
        try:
            number = int(arg)
        except ValueError:
            print("Error: Expected an integer. Received", arg)
            sys.exit(2)
        print(number, "=>", is_prime(number))

File: samples/ruby/04primality/prime.rb
def isPrime(number)
    limit = Math.sqrt(number)
    for c in 2..limit
        if number.modulo(c) == 0 then
            return false
        end
    end
    return true
end

if ARGV.length < 1
    $stderr.puts "Error: Usage `ruby prime.rb <number> [<number> ...]`"
    exit(1)
end

for arg in ARGV
    if (false if Integer(arg) rescue true)
        $stderr.puts "Error: Expected to receive an integer. Received '%s'" % [arg]
        exit(2)
    end

    number = arg.to_i

    puts "%d => %s" % [number, isPrime(number).to_s]
end
